### DISCUSSION TITLE
Build failure on 32bits

### DISCUSSION THREAD

**Discussion by 0323pin** [leftwm/leftwm#1201] [General]
Hi all,

`leftwm` fails to build on NetBSD i386 with the following [error](http://shadow.netbsd.org/pub/pkgsrc/packages/reports/2023Q4/NetBSD-10.0-i386/20231229.0113/leftwm-0.5.1/build.log).

I don't have any 32bits hardware to test but, I thought I should at least mention this.

Thoughts? Do we want to support 32bit systems?

**Comment by Eskaan** [comment]
@0323pin Reading the build log this seems to be a single issue where we try to convert a u32 to a c_long (i64) using the from implementation where a simple type cast `*x as c_long` would have done.

Here's a pr for that: #1202

To test I just installed the rust toolchain for the mentioned hardware (i686-unknown-linux-gnu) and `cargo build --target i686-unknown-linux-gnu` (reproduced and fixed the build error).

**Comment by 0323pin** [comment]
Thanks for the PR 😀
Would you like me to submit it as a patch to our builds and confirm they are fixed?

It should be no problem to merge #1202 

I've found quite a few packages failing on 32bit systems when looking through some build logs yesterday.

Some as this one and some more complex. I got the feeling people no longer care about 32bits or, they simply forget about it.

**Comment by 0323pin** [comment]
@Eskaan Commited as a patch, http://mail-index.netbsd.org/pkgsrc-changes/2023/12/31/msg289844.html
I've requested a pull-up of the fix to the stable branch, so that users using releases can get the fix.

**Reply by Eskaan** [reply]
Great! Thanks :+1:

**Reply by mautamu** [reply]
Thanks Eskaan and 0323pin! 

Merged in 1202 and muted the clippy warning in 1203. For reference, we've previously made a few changes to the code in the past to support 32 bit so I think we should continue to ensure builds work on 32 bit. Thanks for getting this resolved so quickly!
